packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -255,6 +255,11 @@ export function getComponentInternalDef(Ctor: unknown): ComponentDef {
     return def;
 }
 
+export function getComponentHtmlPrototype(Ctor: unknown): HTMLElementConstructor {
```

Comment:
This new API is needed for the `buildCustomElementConstructor` in `@lwc/engine-dom`. It's the only occurrence where a renderer really needs something out of the component def. Instead of exposing the full internal component def, I think that it is safe to only expose the HTML element prototype.